tests/test_reply_function_async.js

```diff
@@ -41,7 +41,11 @@ test('reply takes a callback for status code', async t => {
   const response = await got('http://example.com/')
 
   t.equal(response.statusCode, expectedStatusCode, 'sends status code')
-  t.deepEqual(response.headers, headers, 'sends headers')
+  t.deepEqual(
+    response.headers,
+    { 'x-custom-header': 'abcdef' },
```

Comment:
Hmm, is the casing a change in behavior? I know you called that out elsewhere, though I'm surprised to see the test change.

Reply:
The headers passed from a reply callback were not being processed in the same manner as the headers being passed directly to `.reply`. This is the only place I found `headers` to have a bug.

This is unrelated to the other `rawHeaders` discussion.

Reply:
So `response.headers` is now getting lowercased properly. Gotcha.